File: tests/golden/java-spring-clean-multimodule/core/core/src/main/java/io/github/jtsato/walletservice/core/domains/wallet/usecase/delete/DeleteWalletUseCase.java
package io.github.jtsato.walletservice.core.domains.wallet.usecase.delete;

import io.github.jtsato.walletservice.core.domains.wallet.usecase.delete.DeleteWalletCommand;

public interface DeleteWalletUseCase {
    void execute(DeleteWalletCommand command);
}

File: tests/golden/nestjs-clean-architecture-multimodule/core/packages/core/src/usecases/delete-wallet/delete-wallet.usecase.ts
import { NotFoundException } from '../../exceptions/not-found.exception';
import { DeleteWalletCommand } from './delete-wallet.command';
import { IDeleteWalletGateway } from './delete-wallet.gateway';
import { IDeleteWalletUseCase } from './delete-wallet-usecase.interface';

export class DeleteWalletUseCase implements IDeleteWalletUseCase {
  public constructor(private readonly gateway: IDeleteWalletGateway) {}

  public async execute(command: DeleteWalletCommand): Promise<void> {
    const deleted = await this.gateway.execute(command.id);

    if (!deleted) {
      throw new NotFoundException(`Wallet not found: ${command.id}`);
    }
  }
}
